feat(loki.process): Add forward_to to stage.metrics for direct metric export (#5022)
grafana/alloy: docs/sources/reference/components/loki/loki.process.md

The `stage.metrics` inner block configures stage that allows you to define and update metrics based on values from the shared extracted map.
The created metrics are available at the {{< param "PRODUCT_NAME" >}} root `/metrics` endpoint.

The following arguments are supported:

| Name                    | Type                    | Description                                                    | Default | Required |
| ----------------------- | ----------------------- | -------------------------------------------------------------- | ------- | -------- |
| `forward_to`            | `list(MetricsReceiver)` | Where to forward metrics instead of exposing locally.          |         | no       |
| `metrics_flush_interval`| `duration`              | How frequently to flush metrics to the forwarded receivers.    | `"60s"` | no       |

When `forward_to` is provided, the metrics generated by the `stage.metrics` block are forwarded to the specified receivers (for example, `prometheus.remote_write`) instead of being exposed at the {{< param "PRODUCT_NAME" >}} `/metrics` endpoint.

> This is useful when you want to send derived metrics directly to a remote storage backend without exposing them locally.

Reviewer: ```suggestion
When you provide `forward_to`, the `stage.metrics` block forwards generated metrics to the specified receivers, such as `prometheus.remote_write`, instead of exposing them at the {{< param "PRODUCT_NAME" >}} `/metrics` endpoint.
This allows you to send derived metrics directly to a remote storage backend without exposing them locally.
```

> The `metrics_flush_interval` argument controls how often the metrics are flushed to the receivers specified in `forward_to`. This argument is only used when `forward_to` is set.

Reviewer: ```suggestion
The `metrics_flush_interval` argument controls how often the component flushes metrics to the receivers you specify in `forward_to`. This argument's only used when you set `forward_to`.
```

}
```

The following example shows how to forward metrics to a `prometheus.remote_write` component instead of exposing them at the local `/metrics` endpoint:

```alloy
prometheus.remote_write "default" {
    endpoint {
        url = "http://localhost:9090/api/v1/write"
    }
}

loki.process "example" {
    forward_to = [loki.write.default.receiver]

    stage.regex {
        expression = "^.* order_status=(?P<order_status>.*?) .*$"
    }

    stage.metrics {
        forward_to = [prometheus.remote_write.default.receiver]

        metric.counter {
            name        = "successful_orders_total"
            description = "successful orders"
            source      = "order_status"
            value       = "success"
            action      = "inc"
            prefix      = "loki_log_metrics_"
        }
    }
}
```

> In this example, the `loki_metrics_successful_orders_total` counter metric is sent directly to the remote Prometheus instance via `prometheus.remote_write` instead of being exposed at the {{< param "PRODUCT_NAME" >}} `/metrics` endpoint.

Reviewer: ```suggestion
In this example, `prometheus.remote_write` sends the `loki_log_metrics_successful_orders_total` counter metric directly to the remote Prometheus instance instead of exposing it at the {{< param "PRODUCT_NAME" >}} `/metrics` endpoint.
```
